subsystem/ipv6_test.go

```diff
@@ -0,0 +1,300 @@
+		clusterReply, getErr := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{
+			ClusterID: clusterID,
+		})
+		Expect(getErr).ToNot(HaveOccurred())
+		Expect(len(clusterReply.Payload.HostNetworks)).To(Equal(1))
+		Expect(clusterReply.Payload.HostNetworks[0].Cidr).To(Equal("1001:db8::/120"))
+		By("Installing cluster till finalize")
+		c := installCluster(clusterID)
+		Expect(swag.StringValue(c.Status)).Should(Equal("installing"))
+		Expect(swag.StringValue(c.StatusInfo)).Should(Equal("Installation in progress"))
+		Expect(len(c.Hosts)).Should(Equal(5))
+		for _, host := range c.Hosts {
+			Expect(swag.StringValue(host.Status)).Should(Equal("installing"))
+		}
+
+		for _, host := range c.Hosts {
+			updateProgress(*host.ID, clusterID, models.HostStageDone)
+		}
+
+		waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout, clusterFinalizingStateInfo)
+		By("Completing installation installation")
+		success := true
+		_, err := agentBMClient.Installer.CompleteInstallation(ctx,
+			&installer.CompleteInstallationParams{ClusterID: clusterID, CompletionParams: &models.CompletionParams{IsSuccess: &success, ErrorInfo: ""}})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verifying installation successfully completed")
+		waitForClusterState(ctx, clusterID, models.ClusterStatusInstalled, defaultWaitForClusterStateTimeout, "installed")
+	})
+})
+
+func registerHostsAndSetRolesV6(clusterID strfmt.UUID, numHosts int) []*models.Host {
+	ctx := context.Background()
+	hosts := make([]*models.Host, 0)
+
+	for i := 0; i < numHosts; i++ {
+		hostname := fmt.Sprintf("h%d", i)
+		host := &registerHost(clusterID).Host
+		generateHWPostStepReply(ctx, host, validHwInfoV6, hostname)
+		generateNTPPostStepReply(ctx, host, validNtpSources)
+		var role models.HostRoleUpdateParams
+		if i < 3 {
+			role = models.HostRoleUpdateParamsMaster
+		} else {
+			role = models.HostRoleUpdateParamsWorker
+		}
+		_, err := userBMClient.Installer.UpdateCluster(ctx, &installer.UpdateClusterParams{
+			ClusterUpdateParams: &models.ClusterUpdateParams{HostsRoles: []*models.ClusterUpdateParamsHostsRolesItems0{
+				{ID: *host.ID, Role: role},
+			}},
+			ClusterID: clusterID,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		hosts = append(hosts, host)
+	}
+	generateFullMeshConnectivityV6(ctx, "1001:db8::10", hosts...)
+	apiVip := "1001:db8::64"
+	ingressVip := "1001:db8::65"
+	_, err := userBMClient.Installer.UpdateCluster(ctx, &installer.UpdateClusterParams{
+		ClusterUpdateParams: &models.ClusterUpdateParams{
+			VipDhcpAllocation: swag.Bool(false),
+			APIVip:            &apiVip,
+			IngressVip:        &ingressVip,
+		},
+		ClusterID: clusterID,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	waitForClusterState(ctx, clusterID, models.ClusterStatusReady, 60*time.Second, clusterReadyStateInfo)
+
+	return hosts
+}
+
+func generateFullMeshConnectivityV6(ctx context.Context, startIPAddress string, hosts ...*models.Host) {
```

Comment:
I think we already have a similar function. Couldn't it be reused?

Reply:
Will try